### PR TITLE
feat: Added api endpoints for floors (BREAD)

### PR DESCRIPTION
**Pull request by Anders164a**
## Description

- Added api endpoint for floors (BREAD)
- Added tests for floors (BREAD)

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: src/modules/hotel/hotel.route.ts [hunk 1]
```diff
@@ -75,4 +76,17 @@ export default async (fastify: FastifyInstance, opts: FastifyPluginOptions) => {
...
         deleteHotelHandler
     );
+
+    fastify.get(
+        "/:id/floors",
+        {
+            schema: {
+                tags: ["Hotel"],
+                response: {
+                    204: $ref("browseHotelFloorResponseSchema"),
```
Comment (kristianbinau) [inline src/modules/hotel/hotel.route.ts:86]:
```suggestion
                    200: $ref("browseHotelFloorResponseSchema"),
```

Comment (Anders164a) [reply src/modules/hotel/hotel.route.ts]: Thanks

**Review by kristianbinau — CHANGES_REQUESTED** [review]
Merge master into this and fix all conflics. Try to follow same style and setup as hotels and roomTypes

File: prisma/migrations/20230221080503_/migration.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+-- AlterTable
+ALTER TABLE `Floor` MODIFY `updatedAt` DATETIME(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3);
```
Comment (kristianbinau) [inline prisma/migrations/20230221080503_/migration.sql:2]: Make this in one migration

File: src/modules/floor/floor.service.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    });
+}
+
+export async function findFloorByHotelId(id: number) {
+    return await prisma.floor.findMany({
+        where: {
+            hotelId: id
+        }
+    });
+}
```
Comment (kristianbinau) [inline src/modules/floor/floor.service.ts:74]: Could you edit this and room type as the same as the one in hotel?

File: prisma/migrations/20230221080503_/migration.sql (hunk 2 above)
Comment (Anders164a) [reply prisma/migrations/20230221080503_/migration.sql:2]: Changes added

File: src/modules/floor/floor.schema.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    id: z.number(),
+    ...floorCore
+});
+
+export type CreateFloorInput = z.infer<typeof createFloorSchema>;
+export type UpdateFloorInput = z.infer<typeof updateFloorSchema>;
+export type ShowFloorParams = z.infer<typeof showFloorSchema>;
+export type DeleteFloorParams = z.infer<typeof deleteFloorSchema>;
+
+export const FloorSchema = showFloorResponseSchema;
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/floor/floor.schema.ts:59]: Changes added

File: src/modules/floor/floor.service.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+export async function showFloor(id: number) {
+    const floor = await findFloorById(id);
+    if (!floor) {
+        return idNotFound(id);
+    }
+
+    return floor;
+}
+
+export async function updateFloor(input: UpdateFloorInput) {
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/floor/floor.service.ts:32]: Changes added

File: src/modules/floor/floor.service.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            where: {
+                id: input.id
+            },
+            data: {
+                number: input.number,
+                hotelId: input.hotelId,
+            }
+        })
+
+        return new_floor;
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/floor/floor.service.ts:44]: Changes added

File: src/modules/floor/floor.service.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    } catch (e) {
+        return idNotFound(input.id);
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/floor/floor.service.ts:46]: Changes added

File: src/modules/floor/floor.service.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+export async function deleteFloor(id: number) {
+    try {
+        return await prisma.floor.delete({
+            where: {
+                id: id
+            }
+        });
+    } catch (e) {
+        return idNotFound(id);
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/floor/floor.service.ts:58]: Changes added

File: src/modules/room/type/__test__/updateRoomType.test.ts [hunk 9]
```diff
@@ -10,6 +10,8 @@ describe("POST /api/room/type", () => {
     });
 
     beforeEach(async () => {
+        await fastify.redis.flushall();
+        await prisma.floor.deleteMany();
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/room/type/__test__/updateRoomType.test.ts:14]: Changes added

File: src/modules/room/type/__test__/updateRoomType.test.ts [hunk 10]
```diff
@@ -10,6 +10,8 @@ describe("POST /api/room/type", () => {
     });
 
     beforeEach(async () => {
+        await fastify.redis.flushall();
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/room/type/__test__/updateRoomType.test.ts:13]: Changes added

File: src/modules/floor/floor.service.ts (hunk 3 above)
Comment (Anders164a) [reply src/modules/floor/floor.service.ts:74]: Changes added

File: src/modules/hotel/__test__/deleteHotel.test.ts [hunk 11]
```diff
@@ -10,6 +10,8 @@ describe("DELETE /api/hotel/:id", () => {
     });
 
     beforeEach(async () => {
+        await fastify.redis.flushall();
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/hotel/__test__/deleteHotel.test.ts:13]: Changes added

File: src/modules/hotel/__test__/createHotel.test.ts [hunk 12]
```diff
@@ -10,6 +10,8 @@ describe("POST /api/hotel", () => {
     });
 
     beforeEach(async () => {
+        await fastify.redis.flushall();
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/hotel/__test__/createHotel.test.ts:13]: Changes added

File: src/modules/hotel/__test__/updateHotel.test.ts [hunk 13]
```diff
@@ -10,6 +10,8 @@ describe("POST /api/hotel", () => {
     });
 
     beforeEach(async () => {
+        await fastify.redis.flushall();
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/hotel/__test__/updateHotel.test.ts:13]: Changes added

File: src/modules/room/type/__test__/createRoomType.test.ts [hunk 14]
```diff
@@ -10,6 +10,8 @@ describe("POST /api/room/type", () => {
     });
 
     beforeEach(async () => {
+        await fastify.redis.flushall();
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/room/type/__test__/createRoomType.test.ts:13]: Changes added

File: src/modules/room/type/__test__/browseRoomType.test.ts [hunk 15]
```diff
@@ -11,6 +11,7 @@ describe("GET /api/room/type", () => {
 
     beforeEach(async () => {
         await fastify.redis.flushall();
+        await prisma.floor.deleteMany();
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/room/type/__test__/browseRoomType.test.ts:14]: Changes added

File: src/modules/room/type/__test__/createRoomType.test.ts [hunk 16]
```diff
@@ -10,6 +10,8 @@ describe("POST /api/room/type", () => {
...
+        await prisma.floor.deleteMany();
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/room/type/__test__/createRoomType.test.ts:14]: Changes added

File: src/modules/room/type/__test__/deleteRoomType.test.ts [hunk 17]
```diff
@@ -10,6 +10,8 @@ describe("DELETE /api/room/type/:id", () => {
     });
 
     beforeEach(async () => {
+        await fastify.redis.flushall();
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/room/type/__test__/deleteRoomType.test.ts:13]: Changes added

File: src/modules/room/type/__test__/deleteRoomType.test.ts [hunk 18]
```diff
@@ -10,6 +10,8 @@ describe("DELETE /api/room/type/:id", () => {
...
+        await prisma.floor.deleteMany();
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/room/type/__test__/deleteRoomType.test.ts:14]: Changes added

File: src/modules/room/type/__test__/showRoomType.test.ts [hunk 19]
```diff
@@ -11,6 +11,7 @@ describe("GET /api/room/type/:id", () => {
 
     beforeEach(async () => {
         await fastify.redis.flushall();
+        await prisma.floor.deleteMany();
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/room/type/__test__/showRoomType.test.ts:14]: Changes added

File: src/modules/hotel/hotel.controller.ts [hunk 20]
```diff
@@ -119,4 +121,23 @@ export async function showHotelRoomTypeHandler(
...
+    reply: FastifyReply
+) {
+    try {
+        await request.redis.invalidateCaches(CACHE_KEY_HOTELS);
+        await request.redis.invalidateCaches("allFloors");
+        const floors = await request.redis.rememberJSON<Floor[]>(CACHE_KEY_HOTELS, CACHE_TTL, async () => {
+            return await showHotelFloor(Number(request.params.id));
+        });
+
+        reply.code(200).send(floors);
```
[the remark this answers is not in the document]
Comment (Anders164a) [reply src/modules/hotel/hotel.controller.ts:139]: Whoops

File: src/index.ts [hunk 21]
```diff
@@ -10,27 +10,31 @@ import { roomTypeSchemas } from "./modules/room/type/type.schema";
...
-    });
+        // Register routes
+        await fastify.register(fastifyAutoload, {
+            dir: path.join(__dirname, "./modules/"),
+            options: { prefix: "/api" },
+            matchFilter: (path) => path.endsWith(".route.ts"),
+        });
+    } catch (e) {
+        console.error(e);
+    }
```
Comment (kristianbinau) [inline src/index.ts:37]: remove

File: src/modules/floor/floor.schema.ts [hunk 22]
```diff
@@ -56,7 +56,7 @@ export type UpdateFloorInput = z.infer<typeof updateFloorSchema>;
 export type ShowFloorParams = z.infer<typeof showFloorSchema>;
 export type DeleteFloorParams = z.infer<typeof deleteFloorSchema>;
 
-export const FloorSchema = showFloorResponseSchema;
+export const floorSchema = showFloorResponseSchema;
```
Comment (kristianbinau) [inline src/modules/floor/floor.schema.ts:59]: you forgot to move it between `deleteFloorResponseSchema ` and `CreateFloorInput `

**Review by kristianbinau — CHANGES_REQUESTED** [review]

File: src/modules/floor/floor.schema.ts (hunk 22 above)
Comment (Anders164a) [reply src/modules/floor/floor.schema.ts:59]: Changes added

File: src/index.ts (hunk 21 above)
Comment (Anders164a) [reply src/index.ts:37]: Removed

**Review by kristianbinau — APPROVED** [review]
Awesome!

**Pull request merged**